metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -229,15 +229,21 @@ class DestinationProvider(
     val defnRevisedInput = symbolDefinition.path.toInput
     // Read text file from disk instead of editor buffers because the file
     // on disk is more likely to parse.
-    val parsed =
+    lazy val parsed =
       mtags.index(symbolDefinition.path.toLanguage, defnRevisedInput)
-    if (parsed.occurrences.isEmpty) {
```

Comment:
You can just add `|| symbolDefinition.path.isAmmoniteScript` here

Reply:
Done now (with conditions the other way around).